File: scripts/Filtering_Sorting1.py
total = count = 0.0
infile = open("input.dat","r")
for line in infile:
    try:
        number = int(line)
        if number == 0:
            continue
        total += number
        count += 1
    except ValueError:
        break
print("The average is: %f\n" % (total/count))
infile.close()

File: scripts/Filtering_Sorting1.c
#include <stdio.h>
int main(int argc, char **argv){
int count, total, number; float average;
FILE *infile = fopen("input.dat","r");
for (count = total = 0; fscanf(infile,"%d",&number) != EOF;){
if (number == 0)
continue;
count++; total += number;
}
average = (float)total/(float)count;
printf("The average is: %f\n", average);
fclose(infile);
return 1;
}